storage/src/main/java/com/tencent/rss/storage/handler/impl/HdfsClientReadHandler.java

```diff
@@ -51,7 +66,96 @@ public HdfsClientReadHandler(
         ShuffleStorageUtils.getShuffleDataPathWithRange(appId,
             shuffleId, partitionId, partitionNumPerRange, partitionNum));
     init(fullShufflePath);
-    readAllIndexSegments();
+    cursor = indexReaderMap
+        .entrySet()
+        .stream()
+        .sorted(Comparator.comparing(Entry::getKey))
+        .collect(Collectors.toList())
+        .iterator();
+  }
+
+  public ShuffleIndexResult readShuffleIndex() {
```

Comment:
Don't we read all indexFile like local?

Reply:
Yes, we read a data of a index at once as local mode. But we read index files one by one for there may be multiple index/data files in hdfs mode.

Reply:
Can we read all index file?

Reply:
We must maintain the mapping relation between path and the shuffle index segment if we read all index file first like the first versoin of this pr. And we don't need to read the replica index if all expected blocks have been read.

Reply:
Can we create a class named like 'HdfsShuffleIndexResult' to record the mapping relation.

Reply:
I used to consider this way, but it's kind of overdesign since we have maintain the path to file relation inside the handler. And we would have to add another class HdfsShuffleDataSegment if the class is used.

Reply:
Why is  this overdesign?

Reply:
Both local file and hdfs file has a path. I prefer to have a ShuffleIndexResult with path field than having a HdfsShuffleIndexResult just to add  a common filed, which is implemented in last version.

Reply:
We don't care the path for local file, do it?

Reply:
This is because the relation maintained by the server handler and hdfs file path is maintained by the hdfs read handler now and no need to add a HdfsShuffleIndexResult and HdfsShuffleDataSegment to add path.

Reply:
Could you paste the latest  code?

Reply:
No, I overwrite the last version by current solution since you commented path is appropriate and I agreed with that.

Reply:
Why don't add HdfsShuffleIndexResult? Although we just add a single field, we still have a property abstraction.

Reply:
In this version we don't need it anymore.

Reply:
Actually, I wanna to read all index file, so the mapping relation is necessary.

Reply:
This mapping always needed no matter your read pattern or mode. It's maintained in the handler. And I make it clear at the beginning that I read the whole index file at once and read the file one by one.

Reply:
I hope to read all files? WDYT@colinma